pkg/agent/flowexporter/connections/l7_listener_test.go

```diff
@@ -252,15 +241,37 @@ func TestFlowExporterL7ListenerHttp(t *testing.T) {
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			socketConn, err := net.Dial("unix", l.suricataEventSocketPath)
-			if err != nil {
-				t.Fatalf("Failed to connect to server: %s", err)
-			}
+			timeNow, _ := time.Parse(time.RFC3339Nano, tc.input[0].Timestamp)
+			ctrl := gomock.NewController(t)
+			mockPodStore := podstoretest.NewMockInterface(ctrl)
+			mockPodStore.EXPECT().GetPodByIPAndTime("10.10.0.1", timeNow).AnyTimes().Return(fakeSrcPod, true)
+			mockPodStore.EXPECT().GetPodByIPAndTime("10.10.0.2", timeNow).AnyTimes().Return(fakeDestPod, true)
+			socketFile, err := os.CreateTemp(".", "suricata_test_*.socket")
+			require.NoError(t, err)
+			socketFile.Close()
+			socketPath := socketFile.Name()
+			defer os.RemoveAll(socketPath)
```

Comment:
Maybe it could use `t.TempDir()` which can be automatically removed by Cleanup when test finishes?

Reply:
That's a good idea (would be less verbose), but `t.TempDir()` is a very long path, at least on macOS, and it exceed the max length for a unix socket name.
For example, `/var/folders/1z/5pc5z6b52659pvzq9bz5rm2h0000gn/T/TestFlowExporterL7ListenerHttpInvalid_eventType3141006574/001/suricata_test.socket` is around 150 characters, but the limit it around 100 for both Linux and macOS.

Reply:
I see, thanks for the explanation.